src/tree/AksClusterTreeItem.ts

```diff
@@ -0,0 +1,41 @@
+import { AzExtParentTreeItem, AzureTreeItem, ISubscriptionContext } from "vscode-azureextensionui";
+import { Resource } from "azure-arm-storage/lib/models";
+import { CloudExplorerV1 } from "vscode-kubernetes-tools-api";
+import { SubscriptionClient } from 'azure-arm-resource';
+import { toSubscription } from "../azure-api-utils";
+
+export default class AkClusterTreeItem extends AzureTreeItem {
+    constructor(
+        parent: AzExtParentTreeItem,
+        private readonly resource: Resource) {
+            super(parent);
+
+            // TODO: Set fancy icon for AKS cluster nodes.
+
+            this.id = this.resource.id;
+        }
+
+    public readonly contextValue: string = `aks.cluster ${CloudExplorerV1.SHOW_KUBECONFIG_COMMANDS_CONTEXT}`;
+
+    public get label(): string {
+        return this.name;
+    }
+
+    public get armId(): string {
```

Comment:
Is it worth documenting that these properties exist for compatibility?  I wasn't expecting anyone to build on this so didn't define an API for it, so when DevSpaces did their work they had to use the internal structure at the time as a de facto API.  Now it's no longer obvious what purpose these properties serve, since they just rename base class properties.  A comment explaining about the de facto API might help avoid accidents with downstream consumers in future refactorings/tidyings!

Reply:
Added back the interfaces used in the previous implementation, with a note that it represents a de facto API.